content/stellar-contracts/tokens/vault/vault.mdx

```diff
@@ -0,0 +1,152 @@
+
+```
+assets = (shares × totalAssets) / totalSupply
+```
+
+### Virtual Decimals Offset
+
+The vault uses a "virtual decimals offset" to add extra precision to share calculations.
+This helps prevent rounding errors and improves the accuracy of share-to-asset conversions,
+especially when the vault has few assets or shares.
+
+The offset adds virtual shares and assets to the conversion formula:
+
+```
+shares = (assets × (totalSupply + 10^offset)) / (totalAssets + 1)
+```
+
+The offset is bounded to a maximum of 10 with both security and UX taken into account.
+Values higher than 10 provide minimal practical benefits and may cause overflow errors.
+
+## Rounding Behavior
+
+The vault implements specific rounding behavior to protect the vault from being drained:
+
+- **Deposit/Redeem**: Rounds **down** (depositor receives slightly fewer shares/assets)
+- **Mint/Withdraw**: Rounds **up** (depositor provides slightly more assets/shares)
+
+This ensures the vault always retains a slight advantage in conversions, preventing precision attacks.
+
+| Operation  | Input  | Output | Rounding Direction           |
+| ---------- | ------ | ------ | ---------------------------- |
+| `deposit`  | assets | shares | Down (fewer shares)          |
+| `mint`     | shares | assets | Up (more assets required)    |
+| `withdraw` | assets | shares | Up (more shares burned)      |
+| `redeem`   | shares | assets | Down (fewer assets received) |
+
+## Security Considerations
+
+### Initialization
+
+The vault **MUST** be properly initialized before use:
+
+1. Call `Vault::set_asset(e, asset)` to set the underlying asset
+2. Call `Vault::set_decimals_offset(e, offset)` to set the decimals offset
+3. Initialize metadata with `Base::set_metadata()`
+
+These should typically be done in the constructor. Once set, the asset address and decimals offset are **immutable**.
+
+### Decimal Offset Limits
+
+The decimals offset is limited to a maximum of 10 to prevent:
+- Overflow errors in calculations
+- Excessive precision that provides no practical benefit
+- Poor user experience with unnecessarily large numbers
+
+If a higher offset is required, a custom version of `set_decimals_offset()` must be implemented.
+
+### Inflation (Precision) Attacks
+
+The virtual decimals offset helps protect against inflation attacks where an attacker:
+1. Deposits 1 stroop to get the first share (becoming the sole shareholder)
+2. **Donates** (not deposits) an enormous amount of assets directly to the vault contract via a direct transfer, without receiving any shares in return. This inflates the vault's total assets while keeping total shares at 1, making that single share worth an enormous amount
+3. When a legitimate user tries to deposit a normal amount (e.g., 1000 stroops), the share calculation rounds down to 0 shares because their deposit is negligible compared to the inflated vault balance. The user loses their deposit while receiving nothing
```

Comment:
1000 stroops is still a very small amount. I see it's just for the example, but "normal" is what doesn't make sense here for me.